WordPress/Classes/ViewRelated/Reader/Filter/FilterSheetViewController.swift

```diff
@@ -1,14 +1,86 @@
+import WordPressFlux
+
 class FilterSheetViewController: UIViewController {
 
+    // MARK: Properties
+
     private let viewTitle: String
-    private let filters: [FilterProvider]
+
+    private let filterProvider: FilterProvider
+
+    // closure that's called when a filter item is selected.
     private let changedFilter: (ReaderAbstractTopic) -> Void
 
-    init(viewTitle: String,
-         filters: [FilterProvider],
+    private var receipt: Receipt?
+
+    private var dataSource: FilterTableViewDataSource? {
+        didSet {
+            tableView.dataSource = dataSource
+            tableView.reloadData()
+        }
+    }
+
+    // MARK: Views
+
+    lazy var tableView: UITableView = {
+        let tableView = UITableView()
+        tableView.tableFooterView = UIView() // To hide the separators for empty cells
+        tableView.separatorStyle = .none
+        tableView.delegate = self
+        return tableView
+    }()
+
+    private lazy var emptyView: EmptyActionView = {
+        let view = EmptyActionView(tappedButton: tappedEmptyAddButton)
```

Comment:
This is causing a retain cycle:
![Screenshot 2024-01-02 at 3 12 19 PM](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/4275bb03-92b6-4843-a419-f113caff490b)

```suggestion
        let view = EmptyActionView { [weak self] in
            self?.tappedEmptyAddButton()
        }
```

This may be an existing issue already, since you're copying a lot of the code from the previous file.

Reply:
Whoops, I applied your suggestion but then force pushed from my local changes, causing this to be overwritten... I've re-applied this again here: https://github.com/wordpress-mobile/WordPress-iOS/pull/22273/commits/05504dd2f82d620d8f6b6b774d9d0eeeba7a97f3